### ISSUE TITLE
Create 'Reverse Token' evaluation task

### ISSUE THREAD

**Issue by niznik-dev** [niznik-dev/cruijff_kit#240]
## Summary

Following investigation into why capitalization fine-tuning fails (see `~/.cruijff/2025-12-09_training_inference_mismatch_investigation.md`), we discovered that the capitalization task is **adversarial to how LLMs work** - it requires character-level manipulation but LLMs operate at the token level.

## The Problem with Capitalization

The model sees `picae` as tokens `['p', 'ica', 'e']` and must output `['P', 'ica', 'e']`. There's no structural relationship between token 79 (`p`) and token 47 (`P`) - the model has to memorize mappings rather than learn a generalizable rule. After fine-tuning, the capital letter ranks 200-2800th in probability despite low training loss.

## Proposed: Reverse Token Task

Create a task that operates at the **token level** rather than character level, making it a fairer test of fine-tuning effectiveness.

### Task Definition
- **Input**: A word (e.g., `picae` → tokens `['p', 'ica', 'e']`)
- **Output**: The tokens reversed (e.g., `['e', 'ica', 'p']` → `eicap`)

### Implementation

1. **Dataset generation script** (`data/green/reverse_tokens/generate_reverse_tokens.py`)
   - Reuse input words from capitalization dataset
   - Tokenize each word, reverse the token list, decode back to string
   - Output format: `{"input": "picae", "output": "eicap"}` (or whatever the reversed tokens decode to)
   - Generate train/validation/test splits

2. **Inspect task** (`experiments/reverse_tokens/inspect_task_reverse_tokens.py`)
   - Similar structure to capitalization task
   - Exact match scoring

3. **System prompt** - Something like:
   ```
   Reverse the tokens: apple→elppa|funny→ynnuf|sonic→cinos|...
   ```
   (using actual reversed-token outputs, not character reversal)

### Why This Is Fairer

- The pattern exists at the level the model operates on (tokens)
- Position information is learnable (swap positions 0↔2 for 3-token words)
- No character-level reasoning required
- Tests whether fine-tuning can teach positional/structural transformations

### Acceptance Criteria

- [ ] Dataset generation script that produces train/val/test JSON
- [ ] Inspect task file
- [ ] README documenting the task
- [ ] Verify the task works with scaffold-experiment workflow

## Related

- Capitalization investigation: `/home/niznik/.cruijff/2025-12-09_training_inference_mismatch_investigation.md`
- Original capitalization task: `experiments/capitalization/`

**Comment by niznik-dev** [comment]
I may have experimented with this but I don't recall it leading to anything worthwhile
